File: Spring/Chapter4/java_demo/demo/src/main/java/com/proxyclass/ProxyClassTest.java
package com.proxyclass;

import org.springframework.context.ApplicationContext;
import org.springframework.context.support.ClassPathXmlApplicationContext;

public class ProxyClassTest {
    public static void test()
    {
        ApplicationContext context = new ClassPathXmlApplicationContext("applicationContext.xml");
        ProxyInterface proxyInterface = (ProxyInterface)context.getBean("factory");
        proxyInterface.delete();
        proxyInterface.save();
        proxyInterface.modify();
    }
}

File: Spring/Chapter4/kotlin_demo/demo/src/main/kotlin/proxyclass/ProxyClassTest.kt
package proxyclass

import org.springframework.context.ApplicationContext
import org.springframework.context.support.ClassPathXmlApplicationContext

fun test()
{
    val context:ApplicationContext = ClassPathXmlApplicationContext("applicationContext.xml")
    val proxyClassInterface = context.getBean("factory") as ProxyClassInterface
    proxyClassInterface.delete()
    proxyClassInterface.save()
    proxyClassInterface.modify()
}